### ISSUE TITLE
Impossible to renew a certificate. Works well at creation. 

### ISSUE THREAD

**Issue by jpdupont** [will-in-wi/letsencrypt-webfaction#120]
The cron message give an error and when I try manually, I get this error  :

``` Last login: Fri Mar  9 09:19:37 2018
[alysse@web528 ~]$ ls
bin  certificates  le_config  lib  logs  ruby-2.3.1  src  webapps
[alysse@web528 ~]$ letsencrypt_webfaction --config ~/le_config/config.d55org.yml
/home/alysse/ruby-2.3.1/lib/ruby/2.3.0/fileutils.rb:253:in `mkdir': Permission denied @ dir_s_mkdir - /home/alysse/webapps/d55org (Errno::EACCES)
        from /home/alysse/ruby-2.3.1/lib/ruby/2.3.0/fileutils.rb:253:in `fu_mkdir'
        from /home/alysse/ruby-2.3.1/lib/ruby/2.3.0/fileutils.rb:227:in `block (2 levels) in mkdir_p'
        from /home/alysse/ruby-2.3.1/lib/ruby/2.3.0/fileutils.rb:225:in `reverse_each'
        from /home/alysse/ruby-2.3.1/lib/ruby/2.3.0/fileutils.rb:225:in `block in mkdir_p'
        from /home/alysse/ruby-2.3.1/lib/ruby/2.3.0/fileutils.rb:211:in `each'
        from /home/alysse/ruby-2.3.1/lib/ruby/2.3.0/fileutils.rb:211:in `mkdir_p'
        from /home/alysse/ruby-2.3.1/lib/ruby/gems/2.3.0/gems/letsencrypt_webfaction-2.2.1/lib/letsencrypt_webfaction/domain_validator.rb:43:in `block (2 levels) in write_files!'
        from /home/alysse/ruby-2.3.1/lib/ruby/gems/2.3.0/gems/letsencrypt_webfaction-2.2.1/lib/letsencrypt_webfaction/domain_validator.rb:41:in `each'
        from /home/alysse/ruby-2.3.1/lib/ruby/gems/2.3.0/gems/letsencrypt_webfaction-2.2.1/lib/letsencrypt_webfaction/domain_validator.rb:41:in `block in write_files!'
        from /home/alysse/ruby-2.3.1/lib/ruby/gems/2.3.0/gems/letsencrypt_webfaction-2.2.1/lib/letsencrypt_webfaction/domain_validator.rb:40:in `each'
        from /home/alysse/ruby-2.3.1/lib/ruby/gems/2.3.0/gems/letsencrypt_webfaction-2.2.1/lib/letsencrypt_webfaction/domain_validator.rb:40:in `write_files!'
        from /home/alysse/ruby-2.3.1/lib/ruby/gems/2.3.0/gems/letsencrypt_webfaction-2.2.1/lib/letsencrypt_webfaction/domain_validator.rb:12:in `validate!'
        from /home/alysse/ruby-2.3.1/lib/ruby/gems/2.3.0/gems/letsencrypt_webfaction-2.2.1/lib/letsencrypt_webfaction/application.rb:29:in `run!'
        from /home/alysse/ruby-2.3.1/lib/ruby/gems/2.3.0/gems/letsencrypt_webfaction-2.2.1/exe/letsencrypt_webfaction:5:in `<top (required)>'
        from /home/alysse/.letsencrypt_webfaction/gems/bin/letsencrypt_webfaction:23:in `load'
        from /home/alysse/.letsencrypt_webfaction/gems/bin/letsencrypt_webfaction:23:in `<main>'
[alysse@web528 ~]
```

**Comment by will-in-wi** [comment]
Is `d55org` the correct app folder?

I'd guess that there is a config typo. The failure is coming from the part of the code that creates the authorization file: https://github.com/will-in-wi/letsencrypt-webfaction/blob/93240c438d5c3a28d3b1bea56e9f52488204b8a0/lib/letsencrypt_webfaction/domain_validator.rb#L43

What's happening is that it has successfully contacted Let's Encrypt and obtained the contents of the authorization file. It is now trying to create the folder (presumably something like `/home/alysse/webapps/d55org/.well-known/acme-challenge/something`). In order to do this, it runs a `mkdir -p` on the path leading up to that (`/home/alysse/webapps/d55org/.well-known/acme-challenge`), then writes the file itself.

For some reason, it doesn't see `/home/alysse/webapps/d55org` as existing and tries to create it. This partially makes sense, as WebFaction only allows creation of immediate subfolders of `~/webapps` through the admin UI. That's why this is failing with a permission denied error.

**Comment by jpdupont** [comment]
Yes, there was a file error for d55.
However, I had two other configuration files called in the cron. They work well called alone, but not from the cron, which emailed me the following message:
```
Failed to verify statuses.
/home/alysse/ruby-2.3.1/lib/ruby/gems/2.3.0/gems/letsencrypt_webfaction-2.2.1/lib/letsencrypt_webfaction/domain_validator.rb:67:in `print_error': undefined method `[]' for nil:NilClass (NoMethodError)
	from /home/alysse/ruby-2.3.1/lib/ruby/gems/2.3.0/gems/letsencrypt_webfaction-2.2.1/lib/letsencrypt_webfaction/domain_validator.rb:54:in `block in print_errors'
	from /home/alysse/ruby-2.3.1/lib/ruby/gems/2.3.0/gems/letsencrypt_webfaction-2.2.1/lib/letsencrypt_webfaction/domain_validator.rb:54:in `each'
	from /home/alysse/ruby-2.3.1/lib/ruby/gems/2.3.0/gems/letsencrypt_webfaction-2.2.1/lib/letsencrypt_webfaction/domain_validator.rb:54:in `print_errors'
	from /home/alysse/ruby-2.3.1/lib/ruby/gems/2.3.0/gems/letsencrypt_webfaction-2.2.1/lib/letsencrypt_webfaction/domain_validator.rb:21:in `validate!'
	from /home/alysse/ruby-2.3.1/lib/ruby/gems/2.3.0/gems/letsencrypt_webfaction-2.2.1/lib/letsencrypt_webfaction/application.rb:29:in `run!'
	from /home/alysse/ruby-2.3.1/lib/ruby/gems/2.3.0/gems/letsencrypt_webfaction-2.2.1/exe/letsencrypt_webfaction:5:in `<top (required)>'
	from /home/alysse/.letsencrypt_webfaction/gems/bin/letsencrypt_webfaction:23:in `load'
	from /home/alysse/.letsencrypt_webfaction/gems/bin/letsencrypt_webfaction:23:in `<main>'

```

**Comment by will-in-wi** [comment]
I think you've uncovered a bug in my handling of an edge case. I'm looking into it…

**Comment by will-in-wi** [comment]
I think I've fixed this in v2.2.3. Could you check that this is the case?

**Comment by cbergen** [comment]
I had this same issue. Updated to v2.2.3 and it appears to be working now.
